viewing_party/main.py

```diff
@@ -0,0 +1,218 @@
+            
+
+
+
+
+
+#WAVE 3 
+
+def get_unique_watched(user_data):
+# we need to compare the watched lists for both the
+    # user and the freinds 
+    # find which movies friends haven't watched 
+    # but the user has 
+    # return a list of dict that shows these movies? 
+    # also add in if the list is empty 
+    
+    # user_watched_movies = [] #will hold the value of ["title"]
+    friends_watched_movies = [] #will hold the value of ["title"] 
+    user_unique_titles = [] #movies that we ended up comparing and the user had watched
+    
+    # for movie in user_data["watched"]:
+    #     user_watched_movies.append(movie["title"])       
+
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched_movies.append(movie)
+    
+    
+    #need to loop through unique watched movies 
+    # for unique_movies in user_data["watched"]
+    for unique_movies in user_data["watched"]:
+        if unique_movies not in friends_watched_movies:
+            user_unique_titles.append(unique_movies)
+    
+    return user_unique_titles
+
+
+#  to help if you want to use sets ??  
+# >>> {1, 2, 3, 4} - {2, 3}
+# {1, 4} 
+
+
+
+def get_friends_unique_watched(user_data):
+    # user_watched_movies = []
+    friends_watched_movies = []
+    unique_watched_movies = []
+
+    # for movie in user_data["watched"]:
+    #     user_watched_movies.append(movie["title"])
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched_movies.append(movie)
+    
+
+    
+    for movie in friends_watched_movies:
+        if movie not in user_data["watched"]:
+            if movie not in unique_watched_movies:
+                unique_watched_movies.append(movie)
+    
+    
+    
+    return unique_watched_movies    
+
+        
+    
+    # one friend has watched but the user has NOT watched
+    # for unique_movie in unique_watched_movies: 
+            
+
+
+
+
+#WAVE 4
+
+def get_available_recs(user_data):
+    # movies the user has not watched(can access via their sub) but a friend has watched
+    recommended_movies = []
+
+    # wave 3 function that returns the unique watched movie list
+    # why isn't there a test to check if the user hasn't watched the movie but their friends have? 
+    # unique_movie = get_friends_unique_watched(user_data) 
+    streaming_services = user_data["subscriptions"]
+
+
+
+
+    for friend in user_data["friends"]:
+        for movies in friend.values():
+            for movie in movies: 
+                    if movie["host"] in streaming_services and movie not in recommended_movies:
+                        recommended_movies.append(movie)
+    
+
+
+    return recommended_movies 
+
+
+
+# WAVE 5 
+
+def get_new_rec_by_genre(user_data):
+    rec_movies = []
+    popular_genre = get_most_watched_genre(user_data)
+    unique_movie = get_friends_unique_watched(user_data)
+
+    #   for movie in unique_movie:
```

Comment:
For future reference, its usually best to remove old commented code like this before committing and/or submitting